src/Symfony/Component/Validator/Constraints/JsonValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Imad ZAIRIG <imadzairig@gmail.com>
+ */
+class JsonValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Json) {
+            throw new UnexpectedTypeException($constraint, Json::class);
+        }
+
+        if (null === $value || '' === $value) {
```

Comment:
given #28611 maybe avoid the empty string case for new constraints; could be a trap

Reply:
i want to start that discussion yes :) for now it's consistent :+1: 

more a mention for mergers to consider first if this behavior (#28611) should apply to new constraints

Reply:
@ro0NL I like the changes from #28611 but we should change it all in one go imo. So in this PR this would still be correct, then change it everywhere and remove it in future PRs.